### PR TITLE
SMF crash fix & log cleanup

### PR DESCRIPTION
**Pull request by thakurajayL**

### CODE REVIEW

File: context/sm_context.go [hunk 1]
```diff
@@ -724,7 +724,9 @@ func (smContext *SMContext) PublishSmCtxtInfo() {
 
 	//Populate kafka sm ctxt struct
 	kafkaSmCtxt.Imsi = smContext.Supi
-	kafkaSmCtxt.IPAddress = smContext.PDUAddress.Ip.String()
+	if smContext.PDUAddress.Ip != nil {
+		kafkaSmCtxt.IPAddress = smContext.PDUAddress.Ip.String()
```
Comment (badhrinathpa) [inline context/sm_context.go:728]: in case the IP is nil what will the value of kafkaSmCtxt.IPAddress be ?

Comment (badhrinathpa) [reply context/sm_context.go:728]: Do we need to check PDUAddress also to be nil ?

**Comment by dafATonf** [comment]
test this please

**Comment by gab-arrobo** [comment]
@thakurajayL please rebase your PR

File: context/sm_context.go (hunk 1 above)
Comment (thakurajayL) [reply context/sm_context.go:728]: THanks. Resolved the issue.

File: context/db.go [hunk 2]
```diff
@@ -68,7 +68,7 @@ func SetupSmfCollection() {
 	}
 
 	smfCount := MongoDBLibrary.GetUniqueIdentity("smfCount")
-	logger.CtxLog.Infof("unique id - init smfCount %+v\n", smfCount)
+	logger.CtxLog.Infof("unique id - init smfCount %v", smfCount)
```
Comment (gab-arrobo) [inline context/db.go:71]:
```suggestion
	logger.CtxLog.Infof("unique id - init smfCount %d", smfCount)
```

File: context/db.go [hunk 3]
```diff
@@ -273,7 +273,7 @@ func StoreSeidContextInDB(seidUint uint64, smContext *SMContext) {
 	}
 	itemBsonA := ToBsonMSeidRef(item)
 	filter := bson.M{"seid": seid}
-	logger.CtxLog.Infof("filter : %+v\n", filter)
+	logger.CtxLog.Infof("filter : %+v", filter)
```
Comment (gab-arrobo) [inline context/db.go:276]:
```suggestion
	logger.CtxLog.Infof("filter: %+v", filter)
```

File: context/db.go [hunk 4]
```diff
@@ -287,7 +287,7 @@ func StoreRefToSeidInDB(seidUint uint64, smContext *SMContext) {
 	}
 	itemBsonA := ToBsonMSeidRef(item)
 	filter := bson.M{"ref": smContext.Ref}
-	logger.CtxLog.Infof("filter : %+v\n", filter)
+	logger.CtxLog.Infof("filter : %+v", filter)
```
Comment (gab-arrobo) [inline context/db.go:290]:
```suggestion
	logger.CtxLog.Infof("filter: %+v", filter)
```

File: context/db.go [hunk 5]
```diff
@@ -349,7 +349,7 @@ func DeleteSmContextInDBBySEID(seidUint uint64) {
 	seid := SeidConv(seidUint)
 	fmt.Println("db - delete SMContext In DB by seid")
 	filter := bson.M{"seid": seid}
-	logger.CtxLog.Infof("filter : %+v\n", filter)
+	logger.CtxLog.Infof("filter : %+v", filter)
```
Comment (gab-arrobo) [inline context/db.go:352]:
```suggestion
	logger.CtxLog.Infof("filter: %+v", filter)
```

File: context/db.go [hunk 6]
```diff
@@ -367,7 +367,8 @@ func DeleteSmContextInDBBySEID(seidUint uint64) {
 func DeleteSmContextInDBByRef(ref string) {
 	fmt.Println("db - delete SMContext In DB w ref")
 	filter := bson.M{"ref": ref}
-	logger.CtxLog.Infof("filter : %+v\n", filter)
+	logger.CtxLog.Infof("filter : %+v", filter)
```
Comment (gab-arrobo) [inline context/db.go:370]:
```suggestion
	logger.CtxLog.Infof("filter: %+v", filter)
```

File: context/db_uptunnel.go [hunk 7]
```diff
@@ -171,7 +171,7 @@ func ToBsonMNodeInDB(data *DataPathNodeInDB) (ret bson.M) {
 func StoreNodeInDB(nodeInDB *DataPathNodeInDB) {
 	itemBsonA := ToBsonMNodeInDB(nodeInDB)
 	filter := bson.M{"nodeIDInDB": nodeInDB.DataPathNodeUPFNodeID}
-	logger.CtxLog.Infof("filter : %+v\n", filter)
+	logger.CtxLog.Infof("filter : %+v", filter)
```
Comment (gab-arrobo) [inline context/db_uptunnel.go:174]:
```suggestion
	logger.CtxLog.Infof("filter: %+v", filter)
```

File: context/sm_context.go [hunk 8]
```diff
@@ -724,7 +724,9 @@ func (smContext *SMContext) PublishSmCtxtInfo() {
 
 	//Populate kafka sm ctxt struct
 	kafkaSmCtxt.Imsi = smContext.Supi
-	kafkaSmCtxt.IPAddress = smContext.PDUAddress.Ip.String()
+	if smContext.PDUAddress != nil &&  smContext.PDUAddress.Ip != nil {
```
Comment (gab-arrobo) [inline context/sm_context.go:727]:
```suggestion
	if smContext.PDUAddress != nil && smContext.PDUAddress.Ip != nil {
```

File: pfcp/adapter/adapter.go [hunk 9]
```diff
@@ -198,8 +198,8 @@ func HandlePfcpSessionEstablishmentResponse(msg *pfcpUdp.Message) {
 		}
 	}
 	smContext := smf_context.GetSMContextBySEID(SEID)
-	logger.PfcpLog.Infoln("In HandlePfcpSessionEstablishmentResponse SEID ", SEID)
-	logger.PfcpLog.Infof("In HandlePfcpSessionEstablishmentResponse smContext %+v\n", smContext)
+	logger.PfcpLog.Infof("In HandlePfcpSessionEstablishmentResponse SEID %+v", SEID)
```
Comment (gab-arrobo) [inline pfcp/adapter/adapter.go:201]:
```suggestion
	logger.PfcpLog.Infof("In HandlePfcpSessionEstablishmentResponse SEID %v", SEID)
```

File: pfcp/adapter/adapter.go [hunk 10]
```diff
@@ -235,13 +235,13 @@ func HandlePfcpSessionModificationResponse(msg *pfcpUdp.Message) {
...
-	logger.PfcpLog.Infof("In HandlePfcpSessionModificationResponse pfcpRsp.Cause.CauseValue = [%+v], accepted?? %+v\n", pfcpRsp.Cause.CauseValue, pfcpRsp.Cause.CauseValue == pfcpType.CauseRequestAccepted)
+	logger.PfcpLog.Infof("In HandlePfcpSessionModificationResponse pfcpRsp.Cause.CauseValue = [%v], accepted?? %v", pfcpRsp.Cause.CauseValue, pfcpRsp.Cause.CauseValue == pfcpType.CauseRequestAccepted)
 
 	SEID := msg.PfcpMessage.Header.SEID
-	logger.PfcpLog.Infof("In HandlePfcpSessionModificationResponse SEID %+v\n", SEID)
+	logger.PfcpLog.Infof("In HandlePfcpSessionModificationResponse SEID %v", SEID)
 
 	smContext := smf_context.GetSMContextBySEID(SEID)
-	logger.PfcpLog.Infof("In HandlePfcpSessionModificationResponse smContext found by SEID %+v\n", smContext)
+	logger.PfcpLog.Infof("In HandlePfcpSessionModificationResponse smContext found by SEID %v", smContext)
```
Comment (gab-arrobo) [inline pfcp/adapter/adapter.go:244]:
```suggestion
	logger.PfcpLog.Infof("In HandlePfcpSessionModificationResponse smContext found by SEID %+v", smContext)
```

File: qos/qos_rule_test.go
Comment (gab-arrobo) [inline qos/qos_rule_test.go:1]: Why are the changes in this file needed?

**Review by gab-arrobo — CHANGES_REQUESTED** [review]
Everything looks good. I added a few comments/edits

File: context/sm_context.go (hunk 8 above)
Comment (gab-arrobo) [reply context/sm_context.go:727]: @thakurajayL, FYI, this change fixes the issue with lint

**Comment by thakurajayL** [comment]
retest this please

File: context/sm_context.go (hunk 8 above)
Comment (gab-arrobo) [inline context/sm_context.go:727]:
```suggestion
	if smContext.PDUAddress != nil && smContext.PDUAddress.Ip != nil {
```

File: qos/qos_rule_test.go
Comment (thakurajayL) [reply qos/qos_rule_test.go:1]: Changes correspond to unit test.  Let the changes be there as long as existing tests passes.

**Review by gab-arrobo — APPROVED** [review]
+1

**Review by gab-arrobo — APPROVED** [review]
+1

**Pull request merged**